lib/integrations/jellyfish.ts

```diff
@@ -0,0 +1,52 @@
+import type {
+	Integration,
+	IntegrationDefinition,
+} from '@balena/jellyfish-worker';
+import _ from 'lodash';
+import { hydraAdmin } from '../actions/action-sync-oauth-client';
+
+const SLUG = 'jellyfish';
+
+export class JellyfishIntegration implements Integration {
+	public slug = SLUG;
+	public context: any;
+
+	constructor(options: any) {
+		this.context = options.context;
+	}
+
+	public async destroy() {
+		return;
+	}
+
+	public async translate(): Promise<any> {
+		return [];
+	}
+
+	public async mirror(): Promise<any> {
+		return [];
+	}
+}
+
+export const jellyfishIntegrationDefinition: IntegrationDefinition = {
+	slug: SLUG,
+	initialize: async (options): Promise<JellyfishIntegration> =>
+		new JellyfishIntegration(options),
+	isEventValid: () => true,
+	whoami: async (logContext, credentials) => {
+		const token = await hydraAdmin.introspectOAuth2Token(
+			credentials.access_token,
+		);
+
+		// TODO: Provide sync context instance
+		// TODO: Use elevated worker session instead of kernel.adminSession()
+		return this.context.getElementBySlug(
+			logContext,
+			kernel.adminSession()!,
+			`${token.data.sub}@1.0.0`,
+		);
```

Comment:
@karaxuna We will need to provide access to the `getElementBySlug()` and an elevated session in a similar way that's done for `match()`: https://github.com/product-os/jellyfish/blob/master/apps/server/lib/http/oauth.ts#L52

Reply:
We actually might should just update the first parameter passed to `whoami()` to be an instance of `SyncActionContext` instead of `LogContext`. This is what `match()` has available to it. I don't think this alone will give us access to a privileged session though.

Reply:
Trying this here: https://github.com/product-os/jellyfish-worker/pull/1548

Reply:
Updated the code in this plugin's `whoami()` to use an instance of `SyncActionContext` as mentioned above, which removes the need for direct access to a kernel instance.